packages/back-end/src/routers/teams/teams.controller.ts

```diff
@@ -0,0 +1,262 @@
+  req: AuthRequest,
+  res: Response<GetTeamsResponse>
+) => {
+  const { org } = getOrgFromReq(req);
+
+  req.checkPermissions("manageTeam");
+
+  const teams = await getTeamsForOrganization(org.id);
+
+  return res.status(200).json({
+    status: 200,
+    teams,
+  });
+};
+
+// endregion GET /teams
+
+// region GET /teams/:id
+
+type GetTeamResponse = {
+  status: 200 | 404;
+  team?: TeamInterface;
+  message?: string;
+};
+
+/**
+ * GET /teams/:id
+ * Get team document for the given id
+ * @param req
+ * @param res
+ */
+export const getTeamById = async (
+  req: AuthRequest<null, { id: string }>,
+  res: Response<GetTeamResponse>
+) => {
+  const { org } = getOrgFromReq(req);
+  const { id } = req.params;
+
+  req.checkPermissions("manageTeam");
+
+  const team = await findTeamById(id, org.id);
+  const members = org.members.filter((member) => member.teams?.includes(id));
+  const expandedMembers = await expandOrgMembers(members);
+
+  if (!team) {
+    return res.status(404).json({
+      status: 404,
+      message: "Cannot find team",
+    });
+  }
+
+  return res.status(200).json({
+    status: 200,
+    team: {
+      ...team,
+      members: expandedMembers,
+    },
+  });
+};
+
+// endregion GET /teams/:id
+
+// region PUT /teams/:id
+
+type PutTeamRequest = AuthRequest<
+  {
+    name: string;
+    createdBy: string;
+    description: string;
+    permissions: MemberRoleWithProjects;
+  },
+  { id: string }
+>;
+
+type PutTeamResponse = {
+  status: 200;
+};
+
+/**
+ * PUT /teams/:id
+ * Update team document for the given id
+ * @param req
+ * @param res
+ */
+export const updateTeam = async (
+  req: PutTeamRequest,
+  res: Response<PutTeamResponse>
+) => {
+  const { org } = getOrgFromReq(req);
+  const { name, description, permissions } = req.body;
+  const { id } = req.params;
+
+  req.checkPermissions("manageTeam");
+
+  const team = await findTeamById(id, org.id);
+
+  const changes = await updateTeamMetadata(id, org.id, {
+    name,
+    description,
+    projectRoles: [],
+    ...permissions,
+  });
+
+  await req.audit({
+    event: "team.update",
+    entity: {
+      object: "team",
+      id: id,
+      name: name,
+    },
+    details: auditDetailsUpdate(team, { ...team, ...changes }),
+  });
+
+  return res.status(200).json({
+    status: 200,
+  });
+};
+
+// endregion PUT /teams/:id
+
+// region DELETE /teams/:id
+
+/**
+ * DELETE /teams/:id
+ * Delete team document for given id and remove team id from teams array for any
+ * members of the team.
+ * @param req
+ * @param res
+ */
+export const deleteTeamById = async (
+  req: AuthRequest<null, { id: string }>,
+  res: Response<{ status: 200 }>
+) => {
+  const { org } = getOrgFromReq(req);
+  const { id } = req.params;
+
+  req.checkPermissions("manageTeam");
+
+  const team = await findTeamById(id, org.id);
+
+  const members = org.members.filter((member) => member.teams?.includes(id));
+
+  // Remove members from team to be deleted
+  await Promise.all(
```

Comment:
In some cases in the app, if an asset is being used, we throw a warning and don't let the user delete something until it's unused. E.G We block a user from deleting a metric if the metric is used on an experiment, and instruct them to remove the metric from any experiments. I'm debating whether we want to do that here. I think we do this because we don't have a double confirm delete flow on the front end.